api/src/main/java/org/apache/cloudstack/query/QueryService.java

```diff
@@ -111,8 +111,8 @@
             "allow.user.view.all.domain.accounts", "false",
             "Determines whether users can view all user accounts within the same domain", true, ConfigKey.Scope.Domain);
 
-    static final ConfigKey<Boolean> RestrictPublicTemplatesOfOtherDomains = new ConfigKey<>("Advanced", Boolean.class, "restrict.public.templates.of.other.domains", "false",
-            "If true, the public templates of other domains will not be visible in this domain.", true, ConfigKey.Scope.Domain);
+    static final ConfigKey<Boolean> SharePublicTemplates = new ConfigKey<>("Advanced", Boolean.class, "share.public.templates", "true",
```

Comment:
maybe add more content to configuration name ?

Reply:
What about this name?

```java
    static final ConfigKey<Boolean> SharePublicTemplatesWithOtherDomains = new ConfigKey<>("Advanced", Boolean.class, "share.public.templates.with.other.domains", "true",

````